doc/develop/coding-guidelines.rst

```diff
@@ -0,0 +1,213 @@
+
+.. code-block:: python
+
+    def calculate_mean_and_std(values: List[float]) -> Tuple[float, float]:
+        """
+        Calculates the mean and standard deviation of a list of values
+
+        :param values: list of values.
+        :return: mean and standard deviation
+        """
+
+        mean = sum(values) / len(values)
+        std = () ** 0.5
+        return mean, std
+
+
+In test classes, it is recommended to have one-line docstring for each test method. For short and obvious tests, it is
+not clear if we will use the docstrings.
+
+For e.g. 2.3 the test class should be defined as
+
+.. code-block:: python
+
+    class TestOrsayStatic(unittest.TestCase):
+        """
+        Tests which don't need an Orsay component ready
+        """
+
+        def test_creation(self):
+            """
+            Test to create an Orsay component
+            """
+            if TEST_NOHW == True:
+                self.skipTest(NO_SERVER_MSG)
+            try:
+                oserver = orsay.OrsayComponent(**CONFIG_ORSAY)
+            except Exception as e:
+                self.fail(e)
+            self.assertEqual(len(oserver.children.value), len(CONFIG_ORSAY["children"].keys()))
+
+            oserver.terminate()
+
+        def test_wrong_ip(self):
+            """
+            Tests that an HwError is raised when an empty ip address is entered
+            """
+            with self.assertRaises(HwError):
+                orsay.OrsayComponent(name="Orsay", role="orsay", host="", children="")
+
+
+
+General Guidelines
+
+* For modules, add a license docstring (i.e. within three double quotes) at the top.
+
+  * Exception: It is not necessary for '__init__.py' to carry a license.
+
+* Add the description of the script as a comment below the license.
+
+* Add a docstring to define the class. See example 1.1.
+
+Code Cleaning up
+==================
+
+* If you clean the code for the current feature, make a separate commit for cleaning. The first commit  is recommended for clean-up in the working branch.
+
+* Create a separate pull request if you would like to clean an unrelated code w.r.t current working branch in the repository.
+
+
+Import Order
+==================
+
+Use the following order while importing python modules
+
+* Standard library: See: https://docs.python.org/3/library/
+
+* External: (aka 3rd party) library: e.g. numpy, pandas etc.
+
+* Internal: For e.g. odemis packages.
+
+Within each of the three sub categories follow the alphabetical order. “From” and “import” statements are mixed.
+
+When importing many functions or variables from a single module, the length of the import line may not fit within the maximum line length. In that case, use parentheses to place the import over multiple lines.
+
+For e.g. 4.1 standard and internal imports
+
+.. code-block:: python
+
+    from collections.abc import Iterable, Mapping
+    import cairo
+    import logging
```

Comment:
`import logging` is from the standard lib, so that should go to the first group.